openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/LightGBMUtils.java

```diff
@@ -82,22 +85,33 @@ static CpuArchitecture getCpuArchitecture(final String cpuArchName) {
         }
     }
 
+    static LibcImplementation getLibcImplementation(final String libcImpl) {
+        if (libcImpl == null || libcImpl.isEmpty()) return LibcImplementation.GLIBC;
+
+        try {
+            return LibcImplementation.valueOf(libcImpl.toUpperCase());
+        } catch (final IllegalArgumentException ex) {
+            logger.error("Trying to use LightGBM with an unsupported libc implementation {}.", libcImpl, ex);
+            throw ex;
+        }
+    }
+
     /**
      * Loads a single shared library from the Jar.
      *
      * @param sharedLibResourceName library "filename" inside the jar.
-     * @param cpuArchitecture cpu architecture.
+     * @param infrastructure infrastructure used composed of CPU architecture and libc implementation.
      * @throws IOException if any error happens loading the library.
      */
     static private void loadSharedLibraryFromJar(
             final String sharedLibResourceName,
-            final CpuArchitecture cpuArchitecture
+            final Infrastructure infrastructure
     ) throws IOException {
 
-        logger.debug("Loading LightGBM shared lib: {} for {}.", sharedLibResourceName, cpuArchitecture);
```

Comment:
Can you insert a log with the same semantics please?

Reply:
Added in the latest commit